File: itv/SergioDeIscar-itv-java/SergioDeIscar-itv-java/src/main/java/org/example/models/Vehiculo.java
package org.example.models;

import java.time.LocalDate;

public abstract class Vehiculo {
    private final String matricula;
    private final String marca;
    private final LocalDate anio;
    private final boolean apto;
    private final int kilometro;

    public Vehiculo(String matricula, String marca, LocalDate anio, boolean apto, int kilometro) {
        this.matricula = matricula;
        this.marca = marca;
        this.anio = anio;
        this.apto = apto;
        this.kilometro = kilometro;
    }

    public String getMatricula() {
        return matricula;
    }

    public String getMarca() {
        return marca;
    }

    public LocalDate getAnio() {
        return anio;
    }

    public boolean isApto() {
        return apto;
    }

    public int getKilometro() {
        return kilometro;
    }
}

File: itv/SergioDeIscar-itv-kotlin/SergioDeIscar-itv-kotlin/src/main/kotlin/models/Vehiculo.kt
package models

import java.time.LocalDate

sealed class Vehiculo(val matricula: String, val marca: String, val anio: LocalDate, val apto: Boolean, val kilometro: Int)
class Coche(matricula: String, marca: String, anio: LocalDate, apto: Boolean, kilometro: Int, val numPuertas: Int):
    Vehiculo(matricula, marca, anio, apto, kilometro)
class Moto(matricula: String, marca: String, anio: LocalDate, apto: Boolean, kilometro: Int, val cilindrada: Int):
    Vehiculo(matricula, marca, anio, apto, kilometro)